Adapt batm to Smartcashj

pricky opened the pull request:
- Created two new classes
- Interfaces adapted
- Created some tests that covers ~70%
- gradle adapted for the maven dependency

pricky commented:
Hi, 
Let me know if everything is in a right manner for you.
If you want that I modify something tell me.

Thank GB team

rpanak-generalbytes commented:
Hi, there are a couple of problems with this approach.

1) we don't directly consume artifacts from JitPack. Ideally, we use artifacts from MavenCentral or JCenter, or in case of internal/forked libraries, we use our own repository. We use JitPack only so that contributors can actually build the project without accessing our internal repo. For the final build however, no JitPack-built artifacts are used. This could be solved by smartcash publishing on MavenCentral or JCenter.
2) we're experiencing memory-leak problems with bitcoinj and forks. Because of this we're reluctant to add another bitcoinj fork dependency. We'll have to discuss internally whether we'll allow this. If so, smartcashj would probably need to be based on a very recent bitcoinj version with the (known) leak fixes. Before we proceed we'd like to know whether it is possible for you to implement the functionality without smartcashj dependency.

pricky commented:
> Hi, there are a couple of problems with this approach.
> 
> 1. we don't directly consume artifacts from JitPack. Ideally, we use artifacts from MavenCentral or JCenter, or in case of internal/forked libraries, we use our own repository. We use JitPack only so that contributors can actually build the project without accessing our internal repo. For the final build however, no JitPack-built artifacts are used. This could be solved by smartcash publishing on MavenCentral or JCenter.
> 2. we're experiencing memory-leak problems with bitcoinj and forks. Because of this we're reluctant to add another bitcoinj fork dependency. We'll have to discuss internally whether we'll allow this. If so, smartcashj would probably need to be based on a very recent bitcoinj version with the (known) leak fixes. Before we proceed we'd like to know whether it is possible for you to implement the functionality without smartcashj dependency.

Hi, 
We adapted the point 1.
The point 2 is more difficult to solve. I created some tests that show you what we will use. In this case the functions are light and you can test by yourself .

Let me know what we can do. We need this functionality asap as we ordered several machines... 
thanks

rpanak-generalbytes commented:
Hi, the code seems not to be using the leaking parts of the library. 
However, there is a conflict in one of the files now. Can you please resolve it?

pricky commented:
Hi, wich file is in conflict ? I can resolve ASAP. Let me know what is wrong.

Thank you

rpanak-generalbytes commented:
You can see that just below the comments

pricky commented:
Should I make an other pull request ?

rpanak-generalbytes commented:
I think you messed up the conflict resolution. Take a look at your diff.

pricky commented:
Hi, 
You're right I missed something. It is ok now ?

rpanak-generalbytes commented:
Hi,

the diff still shows some changes that shouldn't be required, like the dependencySubstitutions block in Gradle file.

pricky commented:
Hi,
I moved the content to the file dependencySubstitutions.txt. 
Is it okay for you ??

rpanak-generalbytes commented:
- first of all, I don't think the changes are buildable now (you changed the dependencyChecksums.txt)
- take a look at the diff (https://github.com/GENERALBYTESCOM/bitrafael_public/pull/27/files). Are all of those changes really needed? I don't think so.
- you define a variable with smartcashj version 0.16.0, yet you don't use it anywhere (and use the version 0.17.2 instead)
- what is the reason for all those new dependency substitutions?
I won't look at the code until after these problems are resolved.

pricky commented:
I restored the checksum.
I added only dependencysubstitutions. It will not build without that. 
The version is removed. It was created to respect your version naming. As we only use once the version, I removed that constant.

rpanak-generalbytes commented:
Do you need the jitpack repo definition that you added?

pricky commented:
No, I removed that. I forget the import. Thank you

rpanak-generalbytes commented:
ok, looks much better now. I'd ask you to sort the dependency substitutions so that the same module substitutions are grouped together. I'll take a look at the code today or tomorrow.

pricky commented:
It's done. 
it's perfect, I hope that it will be fine.

rpanak-generalbytes commented on bitrafael-client/src/main/java/com/generalbytes/bitrafael/api/wallet/smart/WalletToolsSMART.java:
```diff
@@ -0,0 +1,265 @@
+        }
+
+        try {
+            Base58.decodeToBigInteger(address);
```
why is this needed?

rpanak-generalbytes commented on bitrafael-client/src/test/java/com/generalbytes/bitrafael/api/wallet/smartcash/SmartcashjTest.java:
```diff
@@ -0,0 +1,94 @@
+import org.junit.Test;
+
+
+public class SmartcashjTest {
```
Tests are insufficient. System.out.println doesn't test anything. Implement proper test cases.

generalbytes commented:
I'm sorry we decided not to have smartcashj as a part of official release of bitrafael.
